src/SimpleWebRequest.ts

```diff
@@ -236,6 +244,33 @@ export abstract class SimpleWebRequestBase<TOptions extends WebRequestOptions =
         }
     }
 
+    private static _scheduleHungRequestCleanupIfNeeded() {
```

Comment:
What about exposing a cleanup function and let the app schedule when to call it? I think not, since this modules knows when the `executingList` changes and can use that to make the scheduling smarter as you have done. Just a thought.

Reply:
Yeah, makes sense.